src/CredentialsWrapper.php

```diff
@@ -141,7 +141,7 @@ public static function build(
         ];
 
         $keyFile = $args['keyFile'];
-        $authHttpHandler = $args['authHttpHandler'] ?: self::buildHttpHandlerFactory();
+        $authHttpHandler = $args['authHttpHandler'];
```

Comment:
nit: let's simplify this and just make the $authHttpHandler supplied to the loader and `CredentialWrapper` come directly from `$args`